demos/object_detection_demo_ssd_async/detection_model.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+// Copyright (C) 2018-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+#include "detection_model.h"
+#include <samples/args_helper.hpp>
+#include <samples/slog.hpp>
+
+using namespace InferenceEngine;
+
+DetectionModel::DetectionModel(std::string modelFileName, float confidenceThreshold, bool useAutoResize, const std::vector<std::string>& labels)
+:ModelBase(modelFileName){
+    this->useAutoResize = useAutoResize;
+    this->confidenceThreshold = confidenceThreshold;
+    this->labels = labels;
+}
+
+void DetectionModel::preprocess(const InputData& inputData, InferenceEngine::InferRequest::Ptr& request, MetaData*& metaData)
```

Comment:
Why is `metaData` passed? You could return it.

Reply:
For consistency. As long as we pass request for filling in, we're also passing metadata same way

Reply:
They are different: `request` was already allocated and you reuse it, but `metaData` was just created.

Reply:
it's still not good idea to return metadata. That would  mean that metadata is main result of this function, and it's not true.

Reply:
I never heard of this interpretation of return value. Well, ok.